lang/java/reef-common/src/main/java/org/apache/reef/runtime/common/driver/evaluator/EvaluatorStatusManager.java

```diff
@@ -43,8 +43,30 @@ private EvaluatorStatusManager() {
   }
 
   private static boolean isLegal(final EvaluatorState from, final EvaluatorState to) {
-    // TODO
-    return true;
+    switch(from) {
+      case ALLOCATED: {
+        if (to == EvaluatorState.SUBMITTED || to == EvaluatorState.FAILED || to == EvaluatorState.KILLED) {
```

Comment:
I'm assuming that an Evaluator with no submitted tasks should never transition to the DONE state. Is this assumption correct?